src/query/service/src/persistent_log/global_persistent_log.rs

```diff
@@ -188,57 +292,102 @@ impl GlobalPersistentLog {
         Ok(())
     }
 
-    async fn prepare(&self) -> Result<()> {
+    pub async fn prepare(&self) -> Result<()> {
         let stage_name = self.stage_name.clone();
         let create_stage = format!("CREATE STAGE IF NOT EXISTS {}", stage_name);
         self.execute_sql(&create_stage).await?;
         let create_db = "CREATE DATABASE IF NOT EXISTS persistent_system";
         self.execute_sql(create_db).await?;
-        let create_table = "
-        CREATE TABLE IF NOT EXISTS persistent_system.query_log (
-            timestamp TIMESTAMP,
-            path VARCHAR,
-            target VARCHAR,
-            log_level VARCHAR,
-            cluster_id VARCHAR,
-            node_id VARCHAR,
-            warehouse_id VARCHAR,
-            query_id VARCHAR,
-            message VARCHAR,
-            fields VARIANT
-        ) CLUSTER BY (timestamp, query_id)";
-        self.execute_sql(create_table).await?;
+
+        for table in &self.tables {
+            let session = create_session(&self.tenant_id, &self.cluster_id).await?;
+            let context = session.create_query_context().await?;
+            let table_name = table.table_name();
+            let old_table = context
+                .get_table(CATALOG_DEFAULT, "persistent_system", table_name)
+                .await;
+            if old_table.is_ok() {
+                let old_schema = old_table?.schema();
```

Comment:
When we rename a table name, what happens to older versions of `databend-query` that are still running?

Reply:
The concern makes sense. The older versions will continue to fail during execution because of schema changes.

However, I think the newer version of databend-query can execute successfully, so logs can still be transferred to the table, although the interval between transfers has become longer.